engine/src/test/java/io/camunda/zeebe/engine/processing/job/JobThrowErrorTest.java

```diff
@@ -387,4 +389,25 @@ public void shouldThrowErrorWithVariablesWithEventSubProcess() {
         .describedAs("With event sub process the variables are created at the process instance")
         .containsExactly(tuple("foo", "\"bar\"", processInstanceKey, VariableIntent.CREATED));
   }
+
+  @Test
+  public void shouldTruncateErrorMessage() {
+    // given
+    final var job = ENGINE.createJob(jobType, PROCESS_ID);
+    final String exceedingErrorMessage = "*".repeat(501);
+
+    // when
+    final Record<JobRecordValue> failedRecord =
+        ENGINE.job().withKey(job.getKey()).withErrorMessage(exceedingErrorMessage).throwError();
+
+    final Record<IncidentRecordValue> incident =
+        RecordingExporter.incidentRecords(CREATED).getFirst();
+
+    // then
+    Assertions.assertThat(failedRecord).hasRecordType(RecordType.EVENT).hasIntent(ERROR_THROWN);
+    assertThat(failedRecord.getValue().getErrorMessage().length()).isEqualTo(500);
+    // this size is greater than 500 because we are re-formatting incident error message by
+    // appending explanation sentences when no catch event found
+    assertThat(incident.getValue().getErrorMessage().length()).isEqualTo(633);
```

Comment:
🔧 We could assert the contents of the error message as well

Reply:
Updated. I removed the length assertions since we are asserting the exact string.